src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -162,6 +162,8 @@ private void ConnectToWorkspace(Workspace workspace)
 
             public void DisconnectFromWorkspace()
             {
+                _reportChangeCancellationSource.Cancel();
```

Comment:
This seems good that it's no longer being nulled out, but note the Connect/Disconnect() pair can happen multiple times in the lifetime of a single tagger. (This is even common: opening a file often goes through an intermediate connection to MiscellaneousFilesWorkspace.) Either this still needs some "create a new token" logic in ConnectToWorkspace, or this really just needs to be in the dispose.

Reply:
➡️ The previous code didn't set to null or dispose. I restricted this change to fix the immediate problem blocking progress on a different change but left the technical debt items unaltered.

Reply:
Oh, I once again forgot this was being set in EnqueueParseSnapshotWorkerAsync. My bad.

Reply:
I guess we have proper one that catch cancellation exception high up the stack?